packages/Meshfree/test/tstDetailsCommunicationHelpers.cpp

```diff
@@ -0,0 +1,172 @@
+/****************************************************************************
+ * Copyright (c) 2012-2017 by the DataTransferKit authors                   *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the DataTransferKit library. DataTransferKit is     *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ ****************************************************************************/
+
+#include <DTK_DetailsDistributedSearchTreeImpl.hpp>   // sendAcrossNetwork
+
+#include <Teuchos_DefaultComm.hpp>
+#include <Teuchos_UnitTestHarness.hpp>
+#include <Tpetra_Distributor.hpp>
+
+template <
+    typename View,
+    typename std::enable_if<
+        std::is_same<
+            typename View::traits::memory_space,
+            typename View::traits::host_mirror_space::memory_space>::value,
+        int>::type = 0>
+Teuchos::ArrayView<typename View::const_value_type> toArray( View const &v )
```

Comment:
Why is this function `const_value_type` and the other `non_const`? Shouldn't keep the constness (const View returns const array and non const View returns non const array)?

Reply:
Nevermind I didn't see that this only a helper for the test.